ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -1088,7 +1089,8 @@ CompletableFuture<Long> getReadIndex() {
 
     // leader has not committed any entries in this term, reject
     if (server.getRaftLog().getTermIndex(readIndex).getTerm() != server.getState().getCurrentTerm()) {
-      return JavaUtils.completeExceptionally(new LeaderNotReadyException(server.getMemberId()));
+      return JavaUtils.completeExceptionally(new ReadIndexException(
+          new LeaderNotReadyException(server.getMemberId())));
```

Comment:
Let's add a message:
```java
      return JavaUtils.completeExceptionally(new ReadIndexException(
          "Failed to getReadIndex " + readIndex + " since the term is not yet committed.",
          new LeaderNotReadyException(server.getMemberId())));
```

Reply:
done